### PR TITLE
Support '[package]' tag in PR CI and fix nightly build on macOS

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
- Support '[package]' tag in PR CI to validate packaging changes in PR CI runs.
  When using '[package]' tag, regular CI tests will run and packaging steps will also run.
- Fix nightly build on macOS.

### CODE REVIEW

**Review by adityapatwardhan — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]

**Review by markekraus — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

File: tools/travis.ps1
Comment (daxian-dbw) [inline tools/travis.ps1]: Maybe it's better to also create a variable `$hasDailyTag`, and use `-or $hasDailyTag` here.

Comment (TravisEz13) [inline tools/travis.ps1]: update docs for new tag

**Review by daxian-dbw — APPROVED** [review]
LGTM except for one comment.

File: tools/travis.ps1
Comment (daxian-dbw) [reply tools/travis.ps1]: Not yet. [Daily] tag is not working for AppVeyor PR CIs yet.

Comment (TravisEz13) [reply tools/travis.ps1]: We could document that it only works in Travis-CI

**Comment by daxian-dbw** [comment]
@SteveL-MSFT Linux CI finished, please check if the package got uploaded.
For the broken macOS CI, #5400 is out trying to fix it.

File: tools/travis.ps1
Comment (SteveL-MSFT) [reply tools/travis.ps1]: ok

**Comment by daxian-dbw** [comment]
For the `[package]` tag, I think it may be unnecessary to run all test like the full build. How about making `[package]` run tests like a regular PR CI but including the package related operation?

**This comment doesn't block the PR.** If we think that's the right way to go, we can submit a separate PR to do it.

**Comment by SteveL-MSFT** [comment]
@daxian-dbw that makes sense since the intent is to test the packaging

File: tools/appveyor.psm1
Comment (daxian-dbw) [inline tools/appveyor.psm1]: We don't need the change in AppVeyor. AppVeyor PR CI always generate packages.
Example: https://ci.appveyor.com/project/PowerShell/powershell/build/6.0.0-beta.9-6723

File: tools/travis.ps1
Comment (daxian-dbw) [inline tools/travis.ps1]: The line `Start-PSBootstrap -Package:(-not $isPr)` needs to be updated.
How about change to:
```powershell
$hasPackageTag = $commitMessage -match '\[package\]'
$createPackages = -not $isPr -or $hasPackageTag
...
Start-PSBootstrap -Package:$createPackages
...
if ($createPackages) {
   ## create package 
```

**Comment by daxian-dbw** [comment]
Since the line `Start-PSBootstrap -Package:(-not $isPr)` is not updated, the build will fail to create the package. I have canceled the Travis CI because of this.

File: tools/appveyor.psm1
Comment (SteveL-MSFT) [reply tools/appveyor.psm1]: got it, will remove

File: tools/travis.ps1
Comment (SteveL-MSFT) [reply tools/travis.ps1]: will update

**Review by daxian-dbw — APPROVED** [review]
LGTM.

**Comment by SteveL-MSFT** [comment]
per @TravisEz13 Travis-CI doesn't upload the artifacts on PRs, so I'll have to wait til this is merged and a daily build kicks off to verify the actual package got uploaded, but it does appear to have been built

**Comment by daxian-dbw** [comment]
It took forever for the macOS CI to start. I will merge this PR without waiting on the macOS because
1. This PR will fix the current build break in macOS -- packaging is broken.
2. The packages are not uploaded for PR CI runs. Merge CI run or daily run will trigger the uploading. @SteveL-MSFT cannot validate the generated arm32 package until this is merged.

**Pull request merged**
